MOBILE_CLIENT/ANDROID/_HIGH/APK_WIFI_API_PII/description.md

```diff
@@ -1,7 +1,72 @@
-The application is using requesting the `ACCESS_WIFI_STATE` interface and calling APIs like `getConnectionInfo` to access sensitive information about the Wi-Fi access point, like BSSID, SSID, and RSSI, and about the device, like MAC address and IP address.
+Android APK WiFi API allows access to Personal Identifiable Information (PII) such as network names and access points usage history, potentially leading to private data inference.
 
-This API is known to be abused to access PII information like:
+Applications using the `ACCESS_WIFI_STATE` permission and calling APIs like `getConnectionInfo` can access sensitive information about WiFi access points (such as `BSSID`, `SSID`, and `RSSI`) and potentially infer:
+
+* MAC Address which is a unique device identifier.
+* Geolocation data by using surrounding WiFi access points.
+* User movement history and social links inference relying on repeating patterns in WiFi access point usage.
+
+### Examples
+
+=== "Java"
+
+ ```java
+ 
+    import android.content.Context;
+    import android.net.wifi.WifiInfo;
+    import android.net.wifi.WifiManager;
+    import java.net.HttpURLConnection;
+    import java.net.URL;
+    import java.io.OutputStream;
+    import java.nio.charset.StandardCharsets;
+    
+    public class WifiInfoRetriever {
+    
+        private Context context;
+    
+        public WifiInfoRetriever(Context context) {
+            this.context = context;
+        }
+    
+        public void retrieveWifiInfo() {
+            WifiManager wifiManager = (WifiManager) context.getSystemService(Context.WIFI_SERVICE);
+            
+            if (wifiManager.isWifiEnabled()) {
+                WifiInfo wifiInfo = wifiManager.getConnectionInfo();
+                
+                String ssid = wifiInfo.getSSID();
+                String bssid = wifiInfo.getBSSID();
+                int rssi = wifiInfo.getRssi();
+                String macAddress = wifiInfo.getMacAddress();
```

Comment:
I think we only care about this part, also in this example the app is not requesting the `ACCESS_WIFI_STATE`

Reply:
As @3asm mentioned, it would nice if we could add an example of this sensitive data is being sent to some external domain, as for `ACCESS_WIFI_STATE`, I think this example sums up the idea pretty well, I do not see it becoming more informative if I add it
What do you think? @ErebusZ